### DISCUSSION TITLE
Generate sectors cache files from sealed

### DISCUSSION THREAD

**Discussion by s0nik42** [filecoin-project/lotus#7314] [Tutorials]
# Disclaimer
This tutorial explains how to  regenerate the cache files from a sealed sector file in case they have been removed or corrupted.
These files are mandatory for wdPoST. 

**/!\ If you have also lost the "p_aux" file located in the same folder. You are unlikely able to get the sector working again.**

- No special permissions requested
- Can be performed on any server 
- 100% CPU (No GPU required)
- Make sure to install rustup from rustup.rs

## Installation
```
git clone https://github.com/filecoin-project/rust-fil-proofs
cd rust-fil-proofs
cargo build --release --bin update_tree_r_cache
```

## Execution 
1. Run
```
./target/release/update_tree_r_cache  rebuild --cache <target_output_folder> --replica <sealed_sector_file> --size <size> # [default: 34359738368]
```
2. Copy the cache folder generated to the correct cache folder
3. Restart the miner
4. Check the deadline to which that sector is attached for verification
```
lotus-miner proving check --slow  0
```

## Example on a 32GiB sector miner 
Execution time on a AMD EPYC 7282 16-Core Processor : 8min
```
./rust-fil-proofs/target/release/update_tree_r_cache rebuild --cache /tmp/cache/s-t010479-2733 --replica sealed/s-t010479-2733
```

# big thanks to @dignifiedquire and @cryptonemo  from PL

**Comment by de-authority** [comment]
It's great doc. thank you very much.
